app/connector/kafka/src/main/java/io/syndesis/connector/kafka/KafkaConnectionCustomizer.java

```diff
@@ -62,29 +63,18 @@ public void customize(ComponentProxyComponent component, Map<String, Object> opt
                 if (!key.isEmpty()) {
                     final String value = attribute.get("value");
                     final String property = getCanonicalPropertyName(key);
-                    try {
-                        //make sure we don't have a specific attribute for this
-                        if (PropertyUtils.isWriteable(configuration, property)) {
-                            Class<?> c = PropertyUtils.getPropertyType(configuration, property);
-                            if(c == Integer.class) {
-                                PropertyUtils.setSimpleProperty(configuration, property, Integer.valueOf(value));
-                            } else if(c == Boolean.class) {
-                                PropertyUtils.setSimpleProperty(configuration, property, Boolean.valueOf(value));
-                            } else {
-                                PropertyUtils.setSimpleProperty(configuration, property, value);
-                            }
-                            PropertyUtils.setSimpleProperty(configuration, property, c.cast(value));
-                        } else {
-                            configuration.getAdditionalProperties().put("additionalProperties." + key,
-                                value);
-                        }
-                    } catch (Exception e) {
-                        LOG.error("Couldn't assign Additional Property " + key, e);
-                        configuration.getAdditionalProperties().put("additionalProperties." + key,
-                            value);
-                    }
+
+                    //In case this is an attribute on KafkaConfiguration that can be set
+                    //So the PropertyBindingSupport does its work below
+                    options.put(property, value);
+
+                    //In case this is an unknown attribute on KafkaConfiguration
+                    configuration.getAdditionalProperties().put("additionalProperties." + property,
+                        value);
                 }
             }
+
+            PropertyBindingSupport.bindProperties(new DefaultCamelContext(), configuration, options);
```

Comment:
+1 for using PropertyBindingSupport, but there is no need to create a new Camel Context. You should instead implement in this class the `CamelContextAware` interface and put there `this.getCamelContext()`. Just as reference see https://github.com/syndesisio/syndesis/blob/master/app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/MongoClientCustomizer.java#L35

Reply:
Solved!